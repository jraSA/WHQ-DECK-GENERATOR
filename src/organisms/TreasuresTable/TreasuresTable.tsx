import React from 'react';
import styled from 'styled-components/native';

import treasuresDeck, { back } from '../../assets/treasuresDeck';
import shuffle from '../../helpers/shuffle/Shuffle';
import Deck from '../../molecules/Deck';
import Table from '../../molecules/Table';

const DeckContainer = styled.View`
    flex: 1;
    flex-direction: row;
    align-self: center;
`;

const TreasuresTable = (props: { navigation: any }) => {
    const renderItems = () => {
        return (
            <DeckContainer>
                <Deck generateDeck={() => shuffle(treasuresDeck)} back={back} />
            </DeckContainer>
        );
    };

    return <Table renderItems={renderItems} navigation={props.navigation} />;
};

export default TreasuresTable;
