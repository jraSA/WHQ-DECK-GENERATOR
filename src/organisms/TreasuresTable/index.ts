import TreasuresTable from './TreasuresTable';

export default TreasuresTable;
