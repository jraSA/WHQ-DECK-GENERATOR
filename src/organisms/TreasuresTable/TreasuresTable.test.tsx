import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import TreasuresTable from './index';

describe('EventsTable Snapshot test', () => {
    it('renders correctly', () => {
        const mockNavigation = jest.fn();
        const tree = renderer.create(<TreasuresTable navigation={mockNavigation} />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
