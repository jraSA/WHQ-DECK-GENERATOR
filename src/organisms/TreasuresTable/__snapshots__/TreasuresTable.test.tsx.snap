// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsTable Snapshot test renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 20,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "center",
                "marginBottom": 40,
                "marginLeft": 40,
                "marginRight": 40,
                "marginTop": 40,
                "width": 233,
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "borderRadius": 22,
                  "height": 360,
                  "width": 233,
                },
              ]
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "borderRadius": 22,
                    "height": 360,
                    "width": 233,
                  },
                  null,
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <View>
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#00609e",
                    "borderRadius": 24,
                    "height": 44,
                    "justifyContent": "center",
                    "margin": 7,
                    "width": 44,
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 360,
                "marginBottom": 40,
                "marginLeft": 40,
                "marginRight": 40,
                "marginTop": 40,
                "width": 233,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "elevation": undefined,
            "justifyContent": "flex-end",
            "zIndex": undefined,
          },
        ],
        undefined,
      ]
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "elevation": undefined,
          "justifyContent": "flex-end",
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": undefined,
        }
      }
    />
    <View
      pointerEvents="box-none"
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "elevation": undefined,
              "justifyContent": "flex-end",
              "zIndex": undefined,
            },
          ],
          Object {
            "alignItems": "flex-end",
          },
          Object {
            "paddingVertical": 30,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginHorizontal": 30,
              "zIndex": undefined,
            },
            Object {
              "elevation": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 5,
                "width": 0,
              },
              "shadowOpacity": 0.35,
              "shadowRadius": 3,
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 96, 158, 1)",
                "borderRadius": 28,
                "height": 56,
                "width": 56,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 28,
                  "height": 56,
                  "justifyContent": "center",
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                  "width": 56,
                }
              }
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "rgba(255, 255, 255, 1)",
                    "fontSize": 24,
                    "marginTop": -4,
                  }
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
